encoding/dot/dotparser/internal/astx/astx.go

```diff
@@ -0,0 +1,322 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Package astx implements utility functions for generating abstract syntax
+// trees of Graphviz DOT graphs.
+package astx
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gonum/graph/encoding/dot/dotparser/ast"
+	"github.com/gonum/graph/encoding/dot/dotparser/internal/token"
+)
+
+// === [ File ] ================================================================
+
+// NewFile returns a new file based on the given graph.
+func NewFile(graph interface{}) (*ast.File, error) {
```

Comment:
Are you expecting that this set of acceptable types will grow? (and below) `interface{}` tends to make me queasy unless strictly necessary.